### PR TITLE
Fix design time dll to build x86 in all flavors but output to the current config outputs

### PR DESCRIPTION
**Pull request by jevansaks**
I was running into a problem when trying to enable nuget package generation on builds that don't produce all flavors. I wanted to get the design dll from an x64 build but the design dll wasn't being generated to the expected location. 

What I did was had the project always build x86 since it's always going to be loaded in 32-bit Visual Studio and it depends on a 32-bit designer dll reference, but it supports all configurations so that it goes to the current configuration output path when it's built.

I also remove the Microsoft.UI.Xaml.vcxproj dependency from the design dll since I didn't see it actually being needed anywhere.

### CODE REVIEW

File: build/CopyFilesToStagingDir.ps1 [hunk 1]
```diff
@@ -32,8 +32,6 @@ PublishFile $FullBuildOutput\Microsoft.UI.Xaml\sdk\Microsoft.UI.Xaml.winmd $Full
 PublishFile $FullBuildOutput\Microsoft.UI.Xaml\Generic.xaml $FullPublishDir\Microsoft.UI.Xaml\
 PublishFile -IfExists $FullBuildOutput\Microsoft.UI.Xaml.Design\Microsoft.UI.Xaml.Design.dll $FullPublishDir\Microsoft.UI.Xaml.Design\
 
-PublishFile -IfExists $BuildOutputDir\$Configuration\AnyCPU\MUXControls.Test.TAEF\MUXControls.Test.dll $FullPublishDir\Test\
-
```
Comment (jevansaks) [inline build/CopyFilesToStagingDir.ps1]: oops, merge casualty

File: dev/Microsoft.UI.Xaml.Design/Microsoft.UI.Xaml.Design.csproj [hunk 2]
```diff
@@ -14,21 +14,39 @@
...
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|arm'">
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug_test|arm'">
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|arm64'">
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|arm64'">
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug_test|arm64'">
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```
Comment (kmahone) [inline dev/Microsoft.UI.Xaml.Design/Microsoft.UI.Xaml.Design.csproj]: Why AllowUnsafeBlocks?

File: dev/Microsoft.UI.Xaml.Design/Microsoft.UI.Xaml.Design.csproj [hunk 3]
```diff
@@ -14,21 +14,39 @@
...
-    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|x86'">
-    <OutputPath>bin\x86\Release\</OutputPath>
-    <PlatformTarget>x86</PlatformTarget>
-    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|x64'">
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug_test|x64'">
```
Comment (kmahone) [inline dev/Microsoft.UI.Xaml.Design/Microsoft.UI.Xaml.Design.csproj]: Can these empty PropertyGroups just be removed? Or does VS want them to be there?

File: dev/Microsoft.UI.Xaml.Design/Microsoft.UI.Xaml.Design.csproj (hunk 2 above)
Comment (jevansaks) [reply dev/Microsoft.UI.Xaml.Design/Microsoft.UI.Xaml.Design.csproj]: VS decided to add that when I was changing configurations, I assume. Will remove.

File: dev/Microsoft.UI.Xaml.Design/Microsoft.UI.Xaml.Design.csproj (hunk 3 above)
Comment (jevansaks) [reply dev/Microsoft.UI.Xaml.Design/Microsoft.UI.Xaml.Design.csproj]: VS needs them there to show them as choices in the configuration manager. Will add a comment.

**Pull request merged**